pkg/relayer/session/session.go

```diff
@@ -225,8 +225,10 @@ func (rp *relayerSessionsManager) validateConfig() error {
 // waitForBlock blocks until the block at the given height (or greater) is
 // observed as having been committed.
 func (rs *relayerSessionsManager) waitForBlock(ctx context.Context, height int64) client.Block {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
 	subscription := rs.blockClient.CommittedBlocksSequence(ctx).Subscribe(ctx)
```

Comment:
Why don't we unsusbscripe anymore? #PUC

Reply:
I think the comment you're asking for already exists in `replayObservable#Subscribe()`:

```
// caller can cancel context or close the publish channel to unsubscribe active observers
```